### ISSUE TITLE
SA1011 false positive with indexed value formatting and string interpolation.

### ISSUE THREAD

**Issue by daviddunson** [DotNetAnalyzers/StyleCopAnalyzers#2961]
```
var bytes = new byte[1];
var generatesWarning = $"{bytes[0]:X2}";
```

Warning SA1011 Closing square bracket must be followed by a space.

**Comment by pantosha** [comment]
@daviddunson, I want to fix it, but I can not reproduce it 😢.
I added the test and test is green:
```csharp
[Fact]
public async Task TestClosingSquareBracketInStringInterpolationWithFormattingAsync()
{
    var testCode = @"public class TestClass
{
    public void TestMethod(object[] a)
    {
        var bytes = new byte[1];
        var generatesWarning = $""{bytes[0]:X2}"";
    }
}
";

    // spacing violation should be reported by SA1013
    await VerifyCSharpDiagnosticAsync(testCode, DiagnosticResult.EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
}
```

**Comment by daviddunson** [comment]
Thank you @pantosha for looking into this. This bug was reported against version 1.0.2 which was the only release version available at the time. It appears to be fixed in the current release (1.1.118).
